<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Summarize, expand, and visualize the data in a TDReport file • GUPPI</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Summarize, expand, and visualize the data in a TDReport file">
<meta property="og:description" content="Reads the data contained in a TDReport file generated by TDPortal or ProSightPD and generates a series of output reports incorporating information from the UniProt knowledge base.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">GUPPI</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0.00</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div id="get-uniprot-protein-info-guppi" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#get-uniprot-protein-info-guppi" class="anchor"></a>Get UniProt Protein Info (GUPPI)</h1></div>
<p><img src="inst/assets/guppilogo.png" width="250"></p>
<p>Process <a href="http://nrtdp.northwestern.edu/tdportal-request/">TDPortal</a> and <a href="http://proteinaceous.net/product/prosightpd/">ProSightPD</a> top-down reports (.tdReport files) by retrieving information from <a href="http://www.uniprot.org/">UniProt</a> and filtering by false detection rate. <a href="http://topdownviewer.northwestern.edu/">TDViewer</a> is also highly recommended for viewing top-down reports.</p>
<p>The easiest way to use the functionality of GUPPI by accessing the <a href="https://icr1.magnet.fsu.edu">GUPPI web application</a>.</p>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>Install from Github with:</p>
<div class="sourceCode" id="cb1"><pre class="r"><span class="kw pkg">remotes</span><span class="kw ns">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html">install_github</a></span>(<span class="st">"davidsbutcher/GUPPI"</span>)</pre></div>
</div>
<div id="input" class="section level2">
<h2 class="hasAnchor">
<a href="#input" class="anchor"></a>Input</h2>
<p>The processing of tdReports is carried out by the <code><a href="reference/guppi.html">guppi()</a></code> function. An example of running the function:</p>
<div class="sourceCode" id="cb2"><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">GUPPI</span>)

<span class="fu"><a href="reference/guppi.html">guppi</a></span>(
   <span class="st">"C:/Users/David Butcher/TDReports"</span>,
   <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(
      <span class="st">"20200420_Excellent_TDReport_01.tdReport"</span>,
      <span class="st">"20200420_Excellent_TDReport_02.tdReport"</span>
   ),
   <span class="fl">83333</span>,
   <span class="kw">GOLocType</span> <span class="kw">=</span> <span class="st">"bacteria"</span>,
   <span class="kw">fractionAssignments</span> <span class="kw">=</span> <span class="kw">NULL</span>,
   <span class="kw">outputdir</span> <span class="kw">=</span> <span class="st">"C:/Users/David Butcher/guppi_output"</span>,
   <span class="kw">fdr</span> <span class="kw">=</span> <span class="fl">0.01</span>,
   <span class="kw">saveOutput</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
   <span class="kw">makeDashboard</span> <span class="kw">=</span> <span class="fl">TRUE</span>
)</pre></div>
<p>Arguments to the <code>guppi</code> function are as follows:</p>
<div id="mandatory-arguments" class="section level3">
<h3 class="hasAnchor">
<a href="#mandatory-arguments" class="anchor"></a>Mandatory arguments</h3>
<ul>
<li><p><code>filedir</code> Add the path to the folder containing tdReports to be processed. This directory can be at any level above the tdReports in the directory structure.</p></li>
<li><p><code>filename</code> Full name of the tdReport file or files, including extension. Must match the name of file exactly.</p></li>
<li><p><code>taxon_number</code> NCBI taxon number to use for adding UniProt data. Should match the taxon number from the TDPortal/ProSightPD analysis.</p></li>
<li><p><code>outputdir</code> Directory to place output files. Directory will be created if it doesn’t exist, but parent directory must be an existing folder.</p></li>
</ul>
</div>
<div id="optional-arguments" class="section level3">
<h3 class="hasAnchor">
<a href="#optional-arguments" class="anchor"></a>Optional arguments</h3>
<ul>
<li><p><code>GOLocType</code> Name of taxon to use for determination of subcellular locations. Acceptable values are “bacteria” or “eukaryota”.</p></li>
<li><p><code>fractionAssignments</code> Optional argument. A named list with names set to the fraction numbers of the input files and values set to the input file names. If left blank, GUPPI will attempt to assign fraction numbers automatically from file names in the tdReport.</p></li>
<li><p><code>fdr</code> False detection rate to use for filtering results. Defaults to 1%.</p></li>
<li><p><code>saveOutput</code> Boolean value. Controls whether protein report, proteoform report, etc. are saved to output directory. Doesn’t effect GUPPI report. Defaults to TRUE.</p></li>
<li><p><code>makeDashboard</code> Boolean value (TRUE or FALSE) which controls whether an HTML report is generated. Defaults to FALSE.</p></li>
<li><p><code>dashboardPath</code> Full path and name of dashboard output (i.e. GUPPI report). Defaults to saving to the /report subdirectory of the output directory.</p></li>
</ul>
</div>
</div>
<div id="analysis-of-tdreports" class="section level2">
<h2 class="hasAnchor">
<a href="#analysis-of-tdreports" class="anchor"></a>Analysis of tdReports</h2>
<p>A connection is established to the SQLite database in the TD Report using <code>RSQLite</code>. All protein-level (or isoform) and proteform-level IDs and other relevant data for each ID are extracted. The taxon number is checked against files in the package directory to see if a corresponding UniProt taxon database has already been downloaded. If not, the UniProt web service is queried for all UniProt accession numbers in the taxon using the package <code>UniProt.ws</code>. Protein name, organism, organism taxon ID, protein sequence, protein function, subcellular location, and any associated GO IDs are returned. Note that some of these values may not be found and come back as empty or NA.</p>
<p>The UniProt taxon database is used to add information for all IDs extracted from the tdReport. GO terms are obtained for all GO IDs using the <code>GO.db</code> package and terms corresponding to subcellular locations are saved in column “GO_subcellular_locations”. Average and monoisotopic masses are determined from the protein sequence (intact sequence for isoform-level IDs and proteoform sequence for proteoform-level IDs) using the <code>Peptides</code> package.</p>
<p>Minimum Q value from among all hits, average and monoisotopic masses, and data file for lowest Q value hit are obtained for all proteoforms. Proteoforms whose Q values are above the FDR cutoff are deleted. Proteoforms whose corresponding protein isoform entry is above the FDR cutoff are also deleted, as in the TDViewer software.</p>
</div>
<div id="output" class="section level2">
<h2 class="hasAnchor">
<a href="#output" class="anchor"></a>Output</h2>
<p>The “main” output includes Q values, observed precursor masses, data files, subcellular locations from the GO database and a variety of other parameters for all protein and proteoform IDs. All protein and proteoform IDs with Q values which are missing or greater than the cutoff value (<code>fdr</code>) are deleted.</p>
<p>Output files are saved to the output directory (<code>outputdir</code>). Files are timestamped with the time the script was initialized or share the same name as the input file.</p>
<ul>
<li>Protein results are saved to <code>outputdir/protein_results/YYYYMMDD_hhmmss_protein_results.xlsx</code>.</li>
<li>Proteoform results are saved to <code>outputdir/proteoform_results/YYYYMMDD_hhmmss_proteoform_results.xlsx</code>.</li>
<li>Output xlsx files contain sheets corresponding to each input tdReport file and a final summary sheet containing all input file names and counts of proteins based on identified subcellular localizations.</li>
<li>Proteoform results including all hits from the tdReport are saved to <code>outputdir/protein_results_allhits/{filename}_allhits.xlsx</code>.</li>
<li>Results showing generalized subcellular localizations determined from protein results for each tdReport are saved to <code>outputdir/protein_results_countsbyfraction/YYYYMMDD_hhmmss_countsbyfrac.xlsx</code>.</li>
<li>If <code>makeDashboard</code> is set to TRUE, an HTML report is saved to <code>outputdir/report</code>.</li>
</ul>
</div>
<div id="dependencies" class="section level2">
<h2 class="hasAnchor">
<a href="#dependencies" class="anchor"></a>Dependencies</h2>
<p>Package dependencies are listed in the <code>Imports</code> section of the DESCRIPTION file and include packages from CRAN, Bioconductor, and Github.</p>
</div>
<div id="license-and-attribution" class="section level2">
<h2 class="hasAnchor">
<a href="#license-and-attribution" class="anchor"></a>License and attribution</h2>
<p>Package developed by David S. Butcher and licensed under <a href="https://creativecommons.org/licenses/by-nc/4.0/">CC BY-NC 4.0</a>.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="https://creativecommons.org/licenses/by-nc/4.0">CC BY-NC 4.0</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>David Butcher <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-1268-4150" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p>Developed by David Butcher.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
